### ISSUE TITLE
Keywords not working: OVERRIDING SYSTEM VALUE 

### ISSUE THREAD

**Issue by johncbranches** [zzzprojects/sqlfiddle4#7]
I was testing the identity generator below and was wondering about the inconvenience that if I deleted a line, that ID could not be added again.
Then I came across the OVERRIDING SYSTEM VALUE which could be used (in theory) to overcome that issue.
However, when I try to execute it on sql fiddle, the following error appears:
"
overriding system value
*
ERROR at line 2:
ORA-03049: SQL keyword 'OVERRIDING' is not syntactically valid following '...,
description) values (2,a,b)
'
"
I asked the SQL fiddle AI chat to check if I was doing anything wrong, but it seems that everything was spelled correctly.

The script I used is below:

-- INIT database
CREATE TABLE Product (
  ProductID INT GENERATED ALWAYS AS IDENTITY (START WITH 1 INCREMENT BY 1),
  Name VARCHAR2(100),
  Description VARCHAR2(255)
);

INSERT INTO Product(Name, Description) VALUES ('Entity Framework Extensions', 'Use <a href="https://entityframework-extensions.net/" target="_blank">Entity Framework Extensions</a> to extend your DbContext with high-performance bulk operations.');
INSERT INTO Product(Name, Description) VALUES ('Dapper Plus', 'Use <a href="https://dapper-plus.net/" target="_blank">Dapper Plus</a> to extend your IDbConnection with high-performance bulk operations.');
INSERT INTO Product(Name, Description) VALUES ('C# Eval Expression', 'Use <a href="https://eval-expression.net/" target="_blank">C# Eval Expression</a> to compile and execute C# code at runtime.');

-- QUERY database
delete from Product
where productid =2;
insert into product (productid, name, description) values (2,a,b)
OVERRIDING SYSTEM VALUE;

SELECT * FROM Product;

**Comment by JonathanMagnan** [comment]
Hello @johncbranches ,

I don't know Oracle well enough, but it looks like that `OVERRIDING SYSTEM VALUE` doesn't exist for ORACLE (this exists for PostgreSQL).

Unfortunately, the AI currently misled us by saying it exists (I got the same answer as you), but I didn't find any online documentation about it.

Best Regards,

Jon

**Comment by johncbranches** [comment]
Hi Jon,

OK, if you say that it is not present in the oracle documentation, then maybe it is just a strange bug from the AI.

What I wanted to do was to find a way to override the GENERATED ALWAYS AS IDENTITY, in case I needed to make backups of the database and insert them again.
(If I backed up and deleted line with ID 1, I could not insert it again from my backup because of the ALWAYS)

Below are the alternative ways of defining a primary key I found and tested successfully to fix this:

1. Sequence(/Sequencer) + trigger (Old way to do things, not relevant for most cases nowadays)
Safer than an automatically generated, because we can insert backups in the Database if needed. NOT TRUE. See further below

CREATE TABLE FACTORY(
ID INT PRIMARY KEY
);

Create sequence factory_id_seq
start with 1
increment by 1
nocache
nocycle;

Create OR REPLACE Trigger factory_id_trigger
before insert ON factory
FOR EACH ROW
BEGIN
IF :NEW.id IS NULL THEN
:NEW.ID := factory_id_seq.NEXTVAL;
END IF;
END;
/


OR 
2. Automatically generated Identity, and modify its generation from always to by default temporarily to add data.

CREATE TABLE Product (
ProductID INT GENERATED ALWAYS AS IDENTITY (START WITH 1 INCREMENT BY 1),
Name VARCHAR2(100),
Description VARCHAR2(255)
);

INSERT INTO Product(Name, Description) VALUES ('entity', 'Use <a href>');
INSERT INTO Product(Name, Description) VALUES ('Dapper Plus', 'target="_blank"');
INSERT INTO Product(Name, Description) VALUES ('C# Eval Expression', 'https://eval-expression.net/');

DELETE FROM PRODUCT
WHERE PRODUCTID=2;

ALTER TABLE PRODUCT MODIFY PRODUCTID INT GENERATED BY DEFAULT AS IDENTITY;
INSERT INTO PRODUCT(PRODUCTID,NAME,dESCRIPTION) VALUES (2,'MEGAPIE','FOO');

ALTER TABLE PRODUCT MODIFY PRODUCTID INT GENERATED ALWAYS AS IDENTITY;

In the previous project I worked such ALTER TABLE commands would not work, as the tables were complex and the command would not run. Some possible reasons for that:

Checklist Before Altering the Table

- Verify that the user has ALTER privilege on the Product table.
- Check for foreign key constraints and disable them if necessary.
- Check for indexes and eventually delete them and rebuild them afterwards
- Ensure there are no active transactions or locks on the Product table.
- Confirm that no triggers or dependent objects will be affected.
- Use tools or queries to check for any active locks or dependencies.

Query for locks

	SELECT object_name, object_type, session_id, locked_mode
	FROM v$locked_object lo
	JOIN dba_objects o ON lo.object_id = o.object_id
	WHERE object_name = 'PRODUCT';
	
Query for disabling and enabling triggers

	ALTER TRIGGER your_trigger DISABLE;
	--ALTER TABLE Product MODIFY ProductID GENERATED BY DEFAULT AS IDENTITY;
	ALTER TRIGGER your_trigger ENABLE;
	
	
OR (Probably the best option)

Create the identity automatically while allowing for lines to be inserted, but keeping the identity unique

CREATE TABLE Product (
  ProductID INT GENERATED BY DEFAULT AS IDENTITY (START WITH 1 INCREMENT BY 1) UNIQUE,
  Name VARCHAR2(100),
  Description VARCHAR2(255)
);

Thank you very much for your quick response Jon :)
